collect_app/src/main/java/org/odk/collect/android/activities/FormMapActivity.kt

```diff
@@ -45,28 +47,30 @@ class FormMapActivity : LocalizedActivity() {
     @Inject
     lateinit var currentProjectProvider: CurrentProjectProvider
 
-    private lateinit var viewModel: FormMapViewModel
+    private val formId by lazy { intent.getLongExtra(EXTRA_FORM_ID, -1) }
+    private val viewModel: FormMapViewModel by viewModels {
+        object : ViewModelProvider.Factory {
+            @Suppress("UNCHECKED_CAST")
+            override fun <T : ViewModel> create(modelClass: Class<T>): T {
+                return FormMapViewModel(
+                    resources,
+                    formId,
+                    formsRepositoryProvider.get(),
+                    instancesRepositoryProvider.get(),
+                    settingsProvider
+                ) as T
+            }
+        }
+    }
 
     public override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
         DaggerUtils.getComponent(this).inject(this)
-        setContentView(R.layout.form_map_activity)
+        supportFragmentManager.fragmentFactory = FragmentFactoryBuilder()
```

Comment:
Using a `FragmentFactory` here seems much cleaner to me than the previous args/service locator approach I'd taken. I've used `FragmentFactory` in a few places, so I'll discuss that here. In general, I think it gives us a really nice solution for passing data to `Fragment` instances. Because the factory is used whenever a fragment is created (both initial `Activity` creation and configuration changes etc) we get the same kind of state retention as arguments but without the need for the objects we're passing to support serialization and be small enough to be written to disk using Android's 1MB buffer. We could look at using this approach as a replacement to Dagger as well (which I think I'd be in favour of) but I didn't want to introduce that change in tactic here. 

In terms of "is this the Android thing to do" I'm less clear there. The newer Fragment docs seem to [enthusiastically suggest providing custom `FragmentFactory` implementations](https://developer.android.com/guide/fragments/fragmentmanager#dependencies). In a contradiction of that however, the [Fragment class documentation](https://developer.android.com/reference/androidx/fragment/app/Fragment#Fragment()) suggests using args instead but without any real justification. 

I'll file an issue with the docs to try and get some clarity on all this but I feel pretty good about using `FragmentFactory` in the way I am here.